### PR TITLE
NET-178: Storage node methods

### PR DESCRIPTION
**Pull request by teogeb**
Methods to query+modify storage nodes. Two new entity classes: `StreamPart` (streamId+partition) and `StorageNode` (currently just an address, but can be extended later).

### CODE REVIEW

**Review by harbu — APPROVED** [review]

File: src/stream/StreamPart.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export default class StreamPart {
+    constructor(streamId, streamPartition) {
+        this._streamId = streamId
+        this.streamPartition = streamPartition
```
[the remark this answers is not in the document]
Comment (teogeb) [reply src/stream/StreamPart.js]: Fixed

File: src/stream/StreamPart.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        const result = []
+        for (let i = 0; i < partitions; i++) {
+            result.push(new StreamPart(id, i))
+        }
+        return result
+    }
+
+    getStreamId() {
+        return this._streamId
+    }
```
Comment (timoxley) [inline src/stream/StreamPart.js:17]: These getters are very "Java style", seems weird, IMO we should probably just expose these as `this.streamId`, `this.streamPartition` or even `this.id` `this.partition`.

No need for getters.

File: src/stream/StreamPart.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        this._streamPartition = streamPartition
+    }
+
+    static fromStream({ id, partitions }) {
+        const result = []
+        for (let i = 0; i < partitions; i++) {
+            result.push(new StreamPart(id, i))
+        }
+        return result
+    }
```
Comment (timoxley) [inline src/stream/StreamPart.js:13]: This is weirdly named, seems like it should be plural at least. `fromX` sounds like it should create a `StreamPart` instance from X, but this gives an array of `StreamPart` instances.

`getStreamPartsFromStream` is a bit long but maybe clearer?

File: src/stream/StorageNode.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export default class StorageNode {
+    constructor(address) {
+        this._address = address
+    }
+
+    getAddress() {
+        return this._address
+    }
+}
```
Comment (timoxley) [inline src/stream/StorageNode.js:9]: Yeah this seems like it shouldn't be a class, or even an object… it's just a string?

**Review by timoxley — COMMENTED** [review]
@teogeb missed this yesterday but added some comments.

**Pull request merged**